Sources/StreamChat/Workers/ChatClientUpdater.swift

```diff
@@ -26,6 +27,7 @@ class ChatClientUpdater {
             client.createBackgroundWorkers()
             // Provide the token to pending API requests
             client.completeTokenWaiters(token: newToken)
+            completion(nil)
```

Comment:
Can we update tests to make sure we don't forget to call completion?

Reply:
`prepareEnvironment` is called from `reloadUserIfNeeded` which had `completion` block already, and it's tested

Reply:
It's some kind of integration tests between 2 funcs. Testing `reloadUserIfNeeded` shouldn't check `prepareEnvironment` behaves as expected, just check it's invoke inside of it. 

Instead, a bunch of tests for `prepareEnvironment ` should be in place to check the behavior. 

Funny fact, it was me who wrote that 😄 Feel free to create a task in tech debt epic to fix it and assign on me 👍

Reply:
CIS-1780 created for this